op-service/client/dial.go

```diff
@@ -62,3 +66,50 @@ func dialRPCClientWithBackoff(ctx context.Context, log log.Logger, addr string)
 		return client, nil
 	})
 }
+
+const BatcherFallbackThreshold int64 = 10
+const ProposerFallbackThreshold int64 = 3
+const TxmgrFallbackThreshold int64 = 3
+
+// DialEthClientWithTimeoutAndFallback will try to dial within the timeout period and create an EthClient.
+// If the URL is a multi URL, then a fallbackClient will be created to add the fallback capability to the client
+func DialEthClientWithTimeoutAndFallback(ctx context.Context, url string, timeout time.Duration, l log.Logger, fallbackThreshold int64, m FallbackClientMetricer) (EthClient, error) {
+	ctxt, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	isMultiUrl, urlList := MultiUrlParse(url)
+	if isMultiUrl {
+		firstEthClient, err := dialRPCClientWithBackoff(ctxt, l, urlList[0])
+		if err != nil {
+			return nil, err
+		}
+		fallbackClient := NewFallbackClient(ethclient.NewClient(firstEthClient), urlList, l, fallbackThreshold, m, func(url string) (EthClient, error) {
+			ctxtIn, cancelIn := context.WithTimeout(ctx, timeout)
+			defer cancelIn()
+			ethClientNew, err := ethclient.DialContext(ctxtIn, url)
+			if err != nil {
+				return nil, err
+			}
+			return ethClientNew, nil
+		})
+		return fallbackClient, nil
+	}
+
+	return DialEthClientWithTimeout(timeout, l, url)
+}
+
+type EthClient interface {
```

Comment:
Doesn't this interface already exist in go-ethereum?

Reply:
Hi, I couldn't find this interface in go-ethereum. I noticed that there is an ETHBackend interface in op-service/txmgr/txmgr.go:63. Also notice that op-node/client/client.go:15 has a Client interface. Neither of these interfaces can be used directly, so I imitated them and wrote my own interface. Could you please tell me the location of the interface in go-ethereum?

Reply:
Are you able to use these interfaces? https://github.com/ethereum/go-ethereum/blob/545f4c5547178bc8bde6af08b3ccaf68ca27f2c0/interfaces.go

Reply:
I can't use this because its interface is too fragmented. For example, in the batcher, both `HeaderByNumber` and `SendTransaction` are required, but they come from two different interfaces. I cannot return multiple interfaces in `DialEthClientWithTimeoutAndFallback`. I need to return a unified EthClient.